src/context.ts

```diff
@@ -265,4 +291,55 @@ export class Context
             this._contextUpdatedEvent.fire();
         }
     }
+
+    public getWorkspaceFolder(path: string) {
+        return this._vscode.getWorkspaceFolder(vscode.Uri.file(path));
+    }
+
+    public async pickPackageType(path : string) {
+        let choices = packages.Type.typePickerChoices();
+        let options: vscode.QuickPickOptions = {
+            placeHolder: 'Select the package type'
+        }
+
+        const chosen = await this._vscode.showQuickPick(choices, options);
+        if (chosen) {
+            this.setPackageType(path, chosen.type);
+        }
+        return chosen;
+    }
+
+    public async pickDebuggingFile(path : string) {
```

Comment:
The debugging target management logic being already there, it seemed to be a good pick. The file picker was generic enough to be there. `pickDebuggingTarget` really does the picking logic (which would be duplicated otherwise) and lets the caller list the choices. I.e. to me, it's utility that definitely has its place here.

Reply:
`pickDebuggingTarget` could be in RockPackage to avoid duplication. But I get your point, let's leave it as you are suggesting..